src/globus_sdk/exc/warnings.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import os
+import typing as t
+import warnings
+
+
+# this is (by meaning) a type of DeprecationWarning, but because python suppresses
+# DeprecationWarning by default, most users would never see it if we inherited from that
+# therefore, inheriting from Warning is an "incorrect" but pragmatic choice
+#
+# in concert with the helpers below to make the warnings opt-in, this results in good
+# user-facing behavior
+class GlobusSDKLegacyBehaviorWarning(Warning):
+    def __init__(
+        self,
+        message: str,
+        *args: t.Any,
+        will_be_removed: str | None = "4.0.0",
```

Comment:
My gut is that there should be no default for the version here; it should always be explicitly passed in at the location the warning is being issued.

It may also be appropriate to disallow `None` values.

Reply:
What if I changed the way that this is declared to be
```
GlobusSDKLegacyBehaviorV4Warning
```
and hardcoded the `4.0.0` as part of the message format?

I don't see any reason to be deprecating things today which we'll remove in any other version. And when we release v4, we can replace this with the V5 variant!

Reply:
Oh, it could also be named more like `GlobusSDKRemovedInV4Warning` or other names like that.

Reply:
I like the no default suggestion.

Hardcoding a specific version of `4.0.0` (either via classname or default string) doesn't save much in terms of keystrokes (the only benefit I can really see) and in return introduces a potential error case.

Reply:
If we were keeping the notion of a warning which could take the version number as an input, I'm convinced that making it a required parameter with no default would be better.

But I've actually come to think that not having any notion of trying to report out the version number in this way is better, at least for now. I'm happy to revisit or discuss further, but for now I'm marking this as resolved.